### ISSUE TITLE
Implementing CAT with GNN

### ISSUE THREAD

**Issue by yogeshhk** [yogeshhk/MidcurveNN#13]
Chordal Axis Transform like approach along with Link Prediction of Graph Neural Network can be combined to generate midcurves.

Training data: Supervised
- Profile and its midcurve data is available
- Triangulate profile shape. Get midpoints of all sides of all triangles. Make a graph.
- In the graph, for 2D profile, only certain links are active (basically the outer shape, for midcurve some other links are active, basically the middle ones.
- Now the problem boils down to, given both inputs and output topologies, can we predict midcurve links given profile graph and links.
-
![shapes_drawio](https://github.com/yogeshhk/MidcurveNN/assets/5088114/0d6346b8-0e2a-4f3c-b996-5a28164aad88)
